### PR TITLE
Added turn_text : shows the current player's turns

### PR DESCRIPTION
**Pull request by jriyyya**
## Overview
Added a text in the activity frame which tells the current player's turn

## Description
I used the TicTacToe activity and noticed that there was no indication of whose turn it was. I reckoned that this was a very common feature for any TicTacToe game and though it'd be nice to have that feature on our sugar activity as well. 
This PR aims to add the aforementioned feature.

## Screenshots
ScreenShot1
Screenshot before the game starts.

ScreenShot2
Screenshot after a turn.

ScreenShot3
Screenshot after a round ends.

## Tests
I have manually used the activity after my modifications and I have not encountered or noticed any user facing bugs in the newly added piece of code.

## Note
The second commit only removes the .vscode folder which is unnecessary and was added by my code editor
The first commit is the feature itself

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -61,6 +61,8 @@ def check_events(self):
                     self.score = [0, 0]
                 if self.help_pos.collidepoint(pg.mouse.get_pos()):
                     self.show_help = not self.show_help
+        self.turn_text = pg.font.Font(None, 64).render(
```
Comment (sparshg) [inline main.py]: This should be placed where other texts are being initialized rather than in `check_events`

File: main.py [hunk 2]
```diff
@@ -181,7 +196,9 @@ def run(self):
                 break
         g.init()
         pg.font.init()
-        self.heading = pg.font.Font(None, 96).render("Tic - Tac - Toe", True, g.WHITE)
+        self.heading = pg.font.Font(None, 96).render(
+            "Tic - Tac - Toe", True, g.WHITE)
+        self.turn_text = pg.font.Font(None, 64).render("X Turn", True, g.WHITE)
```
Comment (sparshg) [inline main.py]: The can be replaced by the line above in `check_events`.

**Review by sparshg — CHANGES_REQUESTED** [review]
Also, the texts and other attributes are calculated every frame while looping inside `run()`. It can be efficient to isolate them so they are only initialized once.

File: main.py (hunk 2 above)
Comment (jriyyya) [reply main.py]: Hi Sparsh, initially this is what I did but the issue with that  approach would be the fact that self.frame isn't initialized within the scope of this block. and since player turn is stored in frame. it would lead to an error of us trying to access an undefined attribute

**Comment by jriyyya** [comment]
Hi @sparshg I have moved it 
the multiple commits were just for testing on my sugar vm. Thanks

File: main.py [hunk 3]
```diff
@@ -73,9 +75,11 @@ def draw_help(self):
             g.WIN.blit(
                 self.close_text,
                 (
-                    (3 * g.WIDTH + g.FRAME_GAP * 3 - 2 * self.close_text.get_width())
+                    (3 * g.WIDTH + g.FRAME_GAP * 3 -
+                     2 * self.close_text.get_width())
                     // 4,
-                    (g.HEIGHT * 0.5 - g.FRAME_GAP * 1.5 - self.close_text.get_height())
+                    (g.HEIGHT * 0.5 - g.FRAME_GAP *
+                     1.5 - self.close_text.get_height())
```
Comment (chimosky) [inline main.py:80]: Why this change?

File: main.py [hunk 4]
```diff
@@ -125,7 +130,16 @@ def draw(self):
             self.heading,
             (
                 (g.WIDTH - self.heading.get_width()) // 2,
-                (g.HEIGHT * 0.5 - g.FRAME_GAP * 1.5 - self.heading.get_height()) // 2,
+                (g.HEIGHT * 0.5 - g.FRAME_GAP * 1.5 -
+                 self.heading.get_height()) // 2,
+            ),
+        )
```
Comment (chimosky) [inline main.py:134]: Why this change?

File: main.py [hunk 5]
```diff
@@ -143,7 +157,8 @@ def draw(self):
         g.WIN.blit(
             scoreo,
             (
-                g.WIDTH - (g.WIDTH - g.FRAME_GAP * 3 + 2 * scorex.get_width()) / 4,
+                g.WIDTH - (g.WIDTH - g.FRAME_GAP * 3 +
+                           2 * scorex.get_width()) / 4,
```
Comment (chimosky) [inline main.py:159]: Why this change?

File: main.py [hunk 6]
```diff
@@ -212,10 +229,12 @@ def run(self):
...
         self.font = pg.font.Font(None, 72)
         self.cross_ui = Animate(self, color=g.ORANGE).cross(
-            ((g.WIDTH - g.FRAME_GAP * 3) / 4, g.HEIGHT / 2 - g.FRAME_GAP / 4), 43, 11
+            ((g.WIDTH - g.FRAME_GAP * 3) / 4,
+             g.HEIGHT / 2 - g.FRAME_GAP / 4), 43, 11
         )
         self.circle_ui = Animate(self, color=g.RED).circle(
-            (g.WIDTH - (g.WIDTH - g.FRAME_GAP * 3) / 4, g.HEIGHT / 2 - g.FRAME_GAP / 4),
+            (g.WIDTH - (g.WIDTH - g.FRAME_GAP * 3) /
+             4, g.HEIGHT / 2 - g.FRAME_GAP / 4),
```
Comment (chimosky) [inline main.py:234]: Why this change?

File: main.py [hunk 7]
```diff
@@ -240,8 +240,8 @@ def run(self):
             self.canvas.grab_focus()
 
         self.frame = Frame(self, (g.WIDTH / 2, g.HEIGHT / 2))
-        self.turn_text = pg.font.Font(None, 64).render(
-            ["O Turn", "", "X Turn"][self.frame.turn+1], True, g.WHITE)
+        # self.turn_text = pg.font.Font(None, 64).render(
+        #     ["O Turn", "", "X Turn"][self.frame.turn+1], True, g.WHITE)
```
Comment (chimosky) [inline main.py]: Delete if they're not needed.

Comment (chimosky) [reply main.py]: Noticed it was deleted in another commit.

**Comment by chimosky** [comment]
Reviewed, not tested and I have some comments;

* You have various commits with the same title, that's not helpful as at first glance it seems you're making the same change over and over again and committing that change, some of the things written in your opening comment should be part of your commit message. They're times when you should have multiple commits but it isn't helpful in this case.
* Also try to avoid merge commits as they make it difficult when viewing diffs with a diff tool, you can rebase when you need to.

You can squash all the commits into one using interactive rebase and then push, let us know if you need any help with that.

File: main.py (hunk 3 above)
Comment (jriyyya) [reply main.py:80]: Fixes inconsistent formatting of the code. It was just for the uniformity of the code. 
Same reason for the rest.

Comment (chimosky) [reply main.py:80]: I don't see the inconsistency you mention and your changes aren't pep8 compliant. 

The title of your change is also for something else entirely so if you want to make any changes to the formatting of the code you should open a different PR for that.

**Comment by jriyyya** [comment]
> * You have various commits with the same title, that's not helpful as at first glance it seems you're making the same change over and over again and committing that change, some of the things written in your opening comment should be part of your commit message. They're times when you should have multiple commits but it isn't helpful in this case.

Sorry about this, Actually I am still figuring my way out for setting up Sugar with ssh, it's a problem with my device.

> * Also try to avoid merge commits as they make it difficult when viewing diffs with a diff tool, you can rebase when you need to
> You can squash all the commits into one using interactive rebase and then push, let us know if you need any help with that.


Will keep these points in my mind for future.

File: main.py (hunk 3 above)
Comment (jriyyya) [reply main.py:80]: Oh Okay, I Will make the code pep8 compliant and open a new PR for that, but what should I do with this PR?

**Comment by chimosky** [comment]
> Sorry about this, Actually I am still figuring my way out for setting up Sugar with ssh, it's a problem with my device.

Oh there's nothing to be sorry about, if you need help please let us know and we'll gladly help.

**Comment by chimosky** [comment]
> Sorry about this, Actually I am still figuring my way out for setting up Sugar with ssh, it's a problem with my device.

I'm curious, what kinda device do you have? What's running on it?

**Comment by jriyyya** [comment]
I'm on a windows 11 laptop with a Sugar Live Build on VirtualBox VM
trying to ssh into the vm with my vs code on windows results in Token Manipulation error

**Comment by chimosky** [comment]
> I'm on a windows 11 laptop with a Sugar Live Build on VirtualBox VM trying to ssh into the vm with my vs code on windows results in Token Manipulation error

Could you copy paste the full error? Do you have OpenSSH installed on your machine?

**Comment by jriyyya** [comment]
> Could you copy paste the full error?

```
user@debian:~$passwd
Changing password for user.
Current password:
passwd: Authentication token manipulation error
passwd: password unchanges
```

when executing `passwd`

> Do you have OpenSSH installed on your machine?

Yes openssh is installed and executing `systemctl status ssh` gives `active (running)`

When trying to ssh from another machine, on the other device it asks the password for user@192.168.xx.xx and after inserting the correct password I get the error of permmission denied (this is what led me to check passwd)

Already tried restarting ssh updating openssh-server and trying different terminals/vscode extensions to ssh into the vm
also the machine is on the bridged adapter so that doesn't appear to be the issue either

**Comment by chimosky** [comment]
> When trying to ssh from another machine, on the other device it asks the password for [user@192.168.xx.xx](mailto:user@192.168.xx.xx) and after inserting the correct password I get the error of permmission denied (this is what led me to check passwd)

I'm guessing you setup a password when you setup the user, what have you tried after getting the error?

Also see if [this](https://www.tecmint.com/fix-passwd-authentication-token-manipulation-error-in-linux/) helps.

**Comment by jriyyya** [comment]
> I'm guessing you setup a password when you setup the user, what have you tried after getting the error?

I even tried using `sudo passwd` where it doesn't ask for the current password to change it, after changing the password and checking for it, I got the same results. 

> Also see if [this](https://www.tecmint.com/fix-passwd-authentication-token-manipulation-error-in-linux/) helps.

I have followed this guide already, and every possible thing available on the internet

**Comment by chimosky** [comment]
Try this and paste the output here, run `ssh -v 0` in the live build environment.

**Comment by jriyyya** [comment]
``` 
user@debian:~$ ssh -v 0
OpenSSH_7.9p1 Debian-10+deb10u2, OpenSSL 1.1.1d  10 Sep 2019
debug1: Reading configuration data /etc/ssh/ssh_config
debug1: /etc/ssh/ssh_config line 19: Applying options for *
debug1: Connecting to 0.0.0.0 [0.0.0.0] port 22.
debug1: Connection established.
debug1: identity file /home/user/.ssh/id_rsa type -1
debug1: identity file /home/user/.ssh/id_rsa-cert type -1
debug1: identity file /home/user/.ssh/id_dsa type -1
debug1: identity file /home/user/.ssh/id_dsa-cert type -1
debug1: identity file /home/user/.ssh/id_ecdsa type -1
debug1: identity file /home/user/.ssh/id_ecdsa-cert type -1
debug1: identity file /home/user/.ssh/id_ed25519 type -1
debug1: identity file /home/user/.ssh/id_ed25519-cert type -1
debug1: identity file /home/user/.ssh/id_xmss type -1
debug1: identity file /home/user/.ssh/id_xmss-cert type -1
debug1: Local version string SSH-2.0-OpenSSH_7.9p1 Debian-10+deb10u2
debug1: Remote protocol version 2.0, remote software version OpenSSH_7.9p1 Debian-10+deb10u2
debug1: match: OpenSSH_7.9p1 Debian-10+deb10u2 pat OpenSSH* compat 0x04000000
debug1: Authenticating to 0.0.0.0:22 as 'user'
debug1: SSH2_MSG_KEXINIT sent
debug1: SSH2_MSG_KEXINIT received
debug1: kex: algorithm: curve25519-sha256
debug1: kex: host key algorithm: ecdsa-sha2-nistp256
debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none
debug1: expecting SSH2_MSG_KEX_ECDH_REPLY
debug1: Server host key: ecdsa-sha2-nistp256 SHA256:N6JEdQfpEZR3aJkbYpT8EYVc9C9MJRXnIT2/TG7u2yo
debug1: Host '0.0.0.0' is known and matches the ECDSA host key.
debug1: Found key in /home/user/.ssh/known_hosts:1
debug1: rekey after 134217728 blocks
debug1: SSH2_MSG_NEWKEYS sent
debug1: expecting SSH2_MSG_NEWKEYS
debug1: SSH2_MSG_NEWKEYS received
debug1: rekey after 134217728 blocks
debug1: Will attempt key: /home/user/.ssh/id_rsa 
debug1: Will attempt key: /home/user/.ssh/id_dsa 
debug1: Will attempt key: /home/user/.ssh/id_ecdsa 
debug1: Will attempt key: /home/user/.ssh/id_ed25519 
debug1: Will attempt key: /home/user/.ssh/id_xmss 
debug1: SSH2_MSG_EXT_INFO received
debug1: kex_input_ext_info: server-sig-algs=<ssh-ed25519,ssh-rsa,rsa-sha2-256,rsa-sha2-512,ssh-dss,ecdsa-sha2-nistp256,ecdsa-sha2-nistp384,ecdsa-sha2-nistp521>
debug1: SSH2_MSG_SERVICE_ACCEPT received
debug1: Authentications that can continue: publickey,password
debug1: Next authentication method: publickey
debug1: Trying private key: /home/user/.ssh/id_rsa
debug1: Trying private key: /home/user/.ssh/id_dsa
debug1: Trying private key: /home/user/.ssh/id_ecdsa
debug1: Trying private key: /home/user/.ssh/id_ed25519
debug1: Trying private key: /home/user/.ssh/id_xmss
debug1: Next authentication method: password
user@0.0.0.0's password: 
debug1: Authentications that can continue: publickey,password
Permission denied, please try again.
user@0.0.0.0's password: 
debug1: Authentications that can continue: publickey,password
Permission denied, please try again.
user@0.0.0.0's password: 
debug1: Authentications that can continue: publickey,password
debug1: No more authentication methods to try.
user@0.0.0.0: Permission denied (publickey,password).
user@debian:~$ 
```

File: main.py (hunk 3 above)
Comment (chimosky) [reply main.py:80]: For this PR you should focus on adding text to show the current player's turn like the subject says so remove every commit that has nothing to do with that.

**Comment by chimosky** [comment]
Nothing stands out, I'll suggest you create a new user with a new password and then try using that user.

File: main.py (hunk 3 above)
Comment (jriyyya) [reply main.py:80]: Okay, I will Do so

Comment (jriyyya) [reply main.py:80]: This PR got really messy, so I will be closing this PR and creating a new clean PR which just includes the necessary commits.
